atlas/atlas.go

```diff
@@ -221,6 +222,52 @@ func (a *Atlas) AddMap(m Map) {
 	a.maps[m.Name] = m
 }
 
+// AddMaps registers maps by name, all or nothing. If a map already exists an error will be returned.
+func (a *Atlas) AddMaps(maps []Map) error {
+	if a == nil {
+		// Use the default Atlas if a, is nil. This way the empty value is
+		// still useful.
+		return defaultAtlas.AddMaps(maps)
+	}
+	a.Lock()
+	defer a.Unlock()
+
+	if a.maps == nil {
+		a.maps = map[string]Map{}
+	}
```

Comment:
since you already know the map is nil here it would be better to:
```go
if a.maps == nil {
   a.maps = make(map[string]Map,len(maps))
   for _, m := range maps {
      a.maps[m.Name] = m
   }
   return nil
}
```

Reply:
Good call. Will do.